src/mantine-core/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  MantineNumberSize,
+  PolymorphicComponentProps,
+  PolymorphicRef,
+  useMantineDefaultProps,
+  ClassNames,
+} from '@mantine/styles';
+import React, { Children, forwardRef, useEffect, useCallback, useState } from 'react';
+import useEmblaCarousel from 'embla-carousel-react';
+import { filterChildrenByType } from '../../utils';
+import { Paper, SharedPaperProps } from '../Paper';
+import { Box } from '../Box';
+import { CarouselItem, CarouselItemStylesName } from './CarouselItem/CarouselItem';
+import useStyles from './Carousel.styles';
+import { CarouselControlProps, DefaultItem } from './DefaultItem/DefaultItem';
+
+export type CarouselStylesName = ClassNames<typeof useStyles> | CarouselItemStylesName;
+
+interface _CarouselProps extends SharedPaperProps {
+  /** Scroll axis of the Carousel items */
+  axis?: 'x' | 'y';
+
+  /** Carousel Items */
+  children: React.ReactNode;
+
+  /** Enables scrolling the carousel through touch or mouse */
+  draggable?: boolean;
+
+  /** If true, the   */
+  scrollOverflow?: boolean;
+
+  /** Group slides together, defaults to 1 */
+  slidesToScroll?: number;
+
+  /** To set the initial scroll item, defaults to the first item i.e 0 */
+  startIndex?: number;
+
+  /** Enable infinite looping the carousel items */
+  shouldLoop?: boolean;
+
+  /** To set the height of the carousel container */
+  height?: MantineNumberSize;
+
+  /** To navigate the carousel */
+  withIndicators?: boolean;
+
+  /** To show/hide the prev and next buttons */
+  withControls?: boolean;
+
+  /** Change prev button component */
+  prevButtonComponent?: React.FC<CarouselControlProps>;
+
+  /** Change next button component */
+  nextButtonComponent?: React.FC<CarouselControlProps>;
+
+  /** Change navigation indicator component */
+  indicatorButtonComponent?: React.FC<CarouselControlProps>;
+}
+
+export type CarouselProps<C> = C extends React.ElementType
+  ? PolymorphicComponentProps<C, _CarouselProps>
+  : never;
+
+type CarouselCompoent = (<C = 'div'>(props: CarouselProps<C>) => React.ReactElement) & {
```

Comment:
I think there is no need for Carousel to be a polymorphic component. It's fine to render just a div as a root element.